File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    vector<string> num = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
    vector<string> digit = { "0", "1", "2", "3", "4", "5", "6", "7", "8", "9" };
    int idx;
    for (int i = 0; i < num.size(); i++) {
        while (1) {
            idx = s.find(num[i]);
            if (idx == string::npos)
                break;
            s.replace(idx, num[i].length(), digit[i]);
        }
    }
    answer = stoi(s);
    return answer;
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
# 숫자의 일부 자릿수가 영단어로 바뀌어졌거나, 혹은 바뀌지 않고 그대로인 문자열 s
# s가 의미하는 원래 숫자를 return
def solution(s):
    arr = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    for i in range(len(arr)):
        idx = s.find(arr[i])
        if idx != -1:
            s = s.replace(arr[i], str(i))            
    
    return int(s)